Assets/Mirror/Components/InterestManagement/SpatialHashing/HexSpatialHash3DInterestManagement.cs

```diff
@@ -0,0 +1,333 @@
+            foreach (NetworkIdentity identity in identityKeys)
+                if (updateStatic || !staticObjects.Contains(identity))
+                    UpdateIdentityPosition(identity); // Refresh cell position for dynamic or scheduled static objects
+
+            if (updateStatic)
+                rebuildCounter = 0; // Reset the counter after updating static objects
+            else
+                rebuildCounter++;
+
+            // Precompute observer sets for each connection before rebuilding
+            connectionObservers.Clear();
+            foreach (NetworkConnectionToClient conn in NetworkServer.connections.Values)
+            {
+                if (conn?.identity == null || !lastConnectionPositions.TryGetValue(conn, out (Cell cell, Vector3 worldPos) connPos))
+                    continue;
+
+                // Get cells visible from the player's position
+                grid.GetNeighborCells(connPos.cell, neighborCells);
+
+                // Initialize the observer set for this connection
+                HashSet<NetworkIdentity> observers = new HashSet<NetworkIdentity>();
+                connectionObservers[conn] = observers;
+
+                // Add all identities in visible cells to the observer set
+                for (int i = 0; i < neighborCells.Length; i++)
+                {
+                    int index = GetCellIndex(neighborCells[i]);
+                    if (index >= 0 && index < cells.Count && cells[index] != null)
+                    {
+                        foreach (NetworkIdentity identity in cells[index])
+                            observers.Add(identity);
+                    }
+                }
+            }
+
+            // RebuildAll invokes NetworkServer.RebuildObservers on all spawned objects
+            base.RebuildAll();
+
+            // Update the last rebuild time
+            lastRebuildTime = NetworkTime.time;
+        }
+    }
+
+    // Called when a new networked object is spawned on the server
+    public override void OnSpawned(NetworkIdentity identity)
+    {
+        // Register the new object's position in the grid system
+        UpdateIdentityPosition(identity);
+
+        // Check if the object is statically batched (indicating it won't move)
+        Renderer[] renderers = identity.gameObject.GetComponentsInChildren<Renderer>();
+        if (renderers.Any(r => r.isPartOfStaticBatch))
+            staticObjects.Add(identity);
+    }
+
+    // Updates the grid cell position of a NetworkIdentity when it moves or spawns
+    void UpdateIdentityPosition(NetworkIdentity identity)
+    {
+        // Get the current world position of the object
+        Vector3 position = identity.transform.position;
+
+        // Convert player position to grid cell coordinates
+        Cell newCell = grid.WorldToCell(position);
+
+        // Check if the object is within �9 km bounds
+        if (Mathf.Abs(position.x) > 9000 || Mathf.Abs(position.y) > 9000 || Mathf.Abs(position.z) > 9000)
```

Comment:
looks like this should be a const? (9000)

Reply:
fixed